application/src/main/java/org/thingsboard/server/service/queue/DefaultTbCoreConsumerService.java

```diff
@@ -592,17 +601,54 @@ private void forwardToStateService(DeviceStateServiceMsgProto deviceStateService
         stateService.onQueueMsg(deviceStateServiceMsg, callback);
     }
 
+    private void forwardToStateService(TransportProtos.DeviceConnectProto deviceConnectMsg, TbCallback callback) {
```

Comment:
I fully agree with you that test is needed and I would appreciate some assistance with this test because I do not fully understand what kind of test do you want. Please message me with more explanation or we can schedule a small meeting. Thanks!